### ISSUE TITLE
Is it going to have a dwm-like configuration?

### ISSUE THREAD

**Issue by Joe23232** [Drakulix/fireplace#56]
With dwm you configure the entire tilling window manager via C code. Is this going to be similar where the configuration is done via your `main.rs` file?

**Comment by Drakulix** [comment]
It won't necessarily.

For window managers like xmonad for example, requiring haskell to dynamically load the configuration file is a non issue, because it is required for running xmonad anyway. So for any interpreted language this is clearly an interesting option, but I have no interest in requiring my users to have a rust toolchain installed on their system, like dwm does (just with a C-toolchain of course).

So there will be (and [already is](https://github.com/Drakulix/fireplace/blob/master/fireplace.yaml)) a yaml configuration file for a barebones fireplace install.

**But** since eventually fireplace shall be used as a base for a gnome-compatible desktop environment, that I want to build, there is a good chance, that fireplace will be split into a library and a binary again. If that works out, it will be possible to just use the library, copy the main.rs and build your own custom fireplace binary. In that case you can just treat the modified main.rs as your config file. (The original main.rs file will not do much more then configuration file parsing in that case anyway.)

If there is enough interest fireplace *might* also support this use-case directly, but that is absolutely uncertain at this point.

**Comment by Joe23232** [comment]
>  If that works out, it will be possible to just use the library, copy the main.rs and build your own custom fireplace binary. In that case you can just treat the modified main.rs as your config file. (The original main.rs file will not do much more then configuration file parsing in that case anyway.)

Yeah I am quite interested in that mate, is it as high level as dwm though?

**Comment by Drakulix** [comment]
I am not invested enough into dwm to compare, but it will be reasonable high-level. I want to abstract away a lot of stuff in fireplace.

**Comment by Joe23232** [comment]
ah I see, thanks mate :)

**Comment by Joe23232** [comment]
@Drakulix Do you provide widget support where we can create widgets or is there another library that does this?
